### PR TITLE
Detect board

### PR DESCRIPTION
**Pull request by vveljko**
Only boards that have a default, intrinsic way of handling
networking (like WiFi boards, such as MKR1000) are recognized.
Also, only boards we could test or that have well-known
implementation and good documentation are recognized.

### CODE REVIEW

**Review by davidnub — COMMENTED** [review]
I'm curious about the `while(true)` statement.

File: examples/AdafruitFeatherM0WINC1500/AdafruitFeatherM0WINC1500.ino [hunk 1]
```diff
@@ -18,63 +17,66 @@ const static char channel[] = "hello_world";
...
-	/* Don't care about the outcome */
-        client->stop();
+  /* Publish */
+  {
+    char msg[] =
+      "\"Hello world from Arduino for Adafruit Feather M0 WINC1500\"";
+    auto client = PubNub.publish(channel, msg);
+    if (!client) {
+      Serial.println("publishing error");
+      delay(1000);
```
[the remark this answers is not in the document]
Comment (vveljko) [reply examples/AdafruitFeatherM0WINC1500/AdafruitFeatherM0WINC1500.ino]: Well, that's just the thing,  sometimes you actually want to make them different. Here they are the same mostly by accident/default. Arduino is infamous for such things, it's not a professional, but a "hobbyist" platform.

File: examples/AdafruitFeatherM0WINC1500/AdafruitFeatherM0WINC1500.ino [hunk 2]
```diff
@@ -18,63 +17,66 @@ const static char channel[] = "hello_world";
...
-            ;
-    }
-    else {
-        Serial.print("WiFi connecting to SSID: ");
-        Serial.println(ssid);
+  // if you're not connected, stop here:
+  if (status != WL_CONNECTED) {
+    Serial.println("Couldn't get a wifi connection");
+    while (true)
+       ;
```
[the remark this answers is not in the document]
Comment (vveljko) [reply examples/AdafruitFeatherM0WINC1500/AdafruitFeatherM0WINC1500.ino]: It's a well-known flame-war how to do this (same line, new line, with `continue`, w/out it, with braces `{}` or w/out...), I just don't care that much any more and just let `clang-format` or some such tool format the code. :)

This is a common theme in Arduino WiFi examples - if you can't connect to WiFi network, report error and "halt". As said elsewhere, such things are common on Arduino, as it's a "hobbyist" environment.

[the remark this answers is not in the document]
Comment (davidnub) [reply examples/AdafruitFeatherM0WINC1500/AdafruitFeatherM0WINC1500.ino]: Ah ok thanks for the information!

**Review by davidnub — APPROVED** [review]
Nice! 👍

**Pull request merged**